storage/remote/write_test.go

```diff
@@ -120,31 +136,51 @@ func TestWriteStorageApplyConfigsPartialUpdate(t *testing.T) {
 		GlobalConfig:       config.GlobalConfig{},
 		RemoteWriteConfigs: []*config.RemoteWriteConfig{c0, c1, c2},
 	}
-	s.ApplyConfig(conf)
+	err = s.ApplyConfig(conf)
+	testutil.Ok(t, err)
 	testutil.Equals(t, 3, len(s.queues))
-	q := s.queues[1]
+	q0, q1, q2 := s.queues[0], s.queues[1], s.queues[2]
 
 	// Update c0 and c2.
-	c0.RemoteTimeout = model.Duration(40 * time.Second)
+	c0.WriteRelabelConfigs[0] = &relabel.Config{Regex: relabel.MustNewRegexp("foo")}
 	c2.RemoteTimeout = model.Duration(50 * time.Second)
 	conf = &config.Config{
 		GlobalConfig:       config.GlobalConfig{},
 		RemoteWriteConfigs: []*config.RemoteWriteConfig{c0, c1, c2},
 	}
-	s.ApplyConfig(conf)
+	err = s.ApplyConfig(conf)
+	testutil.Ok(t, err)
 	testutil.Equals(t, 3, len(s.queues))
 
-	testutil.Assert(t, q == s.queues[1], "Pointer of unchanged queue should have remained the same")
+	testutil.Assert(t, q0 != s.queues[0], "If the index changed, the queue should be stopped and recreated.")
```

Comment:
The message here isn't helpful in my opinion. What index are you referring to? The index of the queue's config in the overall `RemoteWriteConfigs` array? Rearranging those should no longer result in queue restarts. So I think this (and similar lines) should mention which part of the config was changed, and that when changing that field we expect to see a queue restart.